GoogleForCreators/web-stories-wp
# Dashboard: Support deep linking for template details

## Context


Now that Explore Templates shows a modal for the template details, we no longer have specific urls for each template. We would like to be able to continue support to show a specific template's details via url params.
So if you went to `#/templates-gallery?id=59` the template gallery page would load with the modal open to the specific template that matches the id.


## Summary


Route `.../templates-gallery` takes you to the Explore Templates page, whereas a template details page was `templates-gallery/template-detail?id=59&isLocal=false`. Since the modal is replacing the template details route the `/template-detail` portion could be viewed as obsolete. 


## Relevant Technical Choices


- Modal will now add `templateId` and `isLocal` params when opening. ie. `.../template-gallery?id=59&isLocal-false`.
- Decided to make the new functionally backwards compatible via redirecting any nested routes along with the params back to `template-gallery`. ie. `.../template-gallery/template-detail?id={templateId}` -> `.../template-gallery?id={templateId}`. 
- Removed `centerTargetAction` as part of the  `reshapedTemplatedObject` since this was used for the relatedTemplates and as a link to a specific template details page.  Though since we do have backwards compatibly and it could be useful to link to a specific template, I think perhaps I was haste in removing this. Thoughts?
- 

## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Go to the explore templates page and click 'See Details' on any template
2. Modal should open with new params in the url. ie `#/templates-gallery?id=52&isLocal=false`
3. Copy url.
4. Close modal and the params should be removed
5. Navigate away from explore templates page
6.  Paste in url that contains the template id param and check that you are taken to the explore templates page with the modal open to the appropriate template details. 
7. You can also check that old url work by adding in the `template-detail` portion eg.  `/templates-gallery/template-detail?id=52&isLocal=false`


## Reviews

### Does this PR have a security-related impact?
no


### Does this PR change what data or activity we track or use?
yes


### Does this PR have a legal-related impact?
no


## Checklist




---



Fixes #9996
